bin/file-mod

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+#
+# Copyright:: Copyright 2017 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -eou pipefail
+
+function usage() {
+  cat <<DOC
+Usage: ${0##*/} SUBCOMMAND [PARAMETERS]
+
+SUBCOMMANDS:
+    append-if-missing STRING FILE             Append STRING to FILE if not already there.
+    find-and-replace REGEX_STR STRING FILE    Replace REGEX_STR with STRING in FILE. Supports multiline replace. Uses perl.
+DOC
+}
+
+function append_if_missing() {
+  string="$2"
```

Comment:
A nice idiom to make both the code and usage more understandable:
```bash
  string=${2:?missing parameter: string to append}
```
Changes the error on usage from:
```
➤ bash bin/file-mod  append-if-missing
bin/file-mod: line 32: $2: unbound variable
```
to
```
bin/file-mod: line 32: 2: missing parameter: string to append
```

Also, double quotes aren't necessary in assignment statements (without glob characters) since the RHS isn't subject to word-splitting. But it doesn't hurt to have them either.